lms/djangoapps/grades/models.py

```diff
@@ -107,7 +107,8 @@ def hash_from_blockrecords(self, blocks):
             blocks = BlockRecordSet(blocks)
 
         try:
-            model = self.create_from_blockrecords(blocks)
+            with transaction.atomic():
+                model = self.create_from_blockrecords(blocks)
```

Comment:
@sanfordstudent These additional `with transaction.atomic()` lines have coincidentally brought the number of queries in tests back to where they were before. So the net result is the same number of queries, but with added safety in case of database errors by using django best practices.